test/csuite/wt8963_insert_stress/main.c

```diff
@@ -0,0 +1,171 @@
+ * The test creates a lot of threads that concurrently insert a lot of records with random keys.
+ * Having a large memory page ensures that we have big insert lists. Big cache size allows having
+ * more dirty content in the memory before eviction kicks in.
+ */
+
+#define THREAD_NUM_ITERATIONS 200000
+#define NUM_THREADS 110
+#define KEY_MAX UINT32_MAX
+
+void *thread_insert_race(void *);
+
+static uint64_t ready_counter;
+static WT_RAND_STATE rnd;
+
+/*
+ * set_key --
+ *     Wrapper providing the correct typing for the WT_CURSOR::set_key variadic argument.
+ */
+static void
+set_key(WT_CURSOR *c, uint64_t value)
+{
+    c->set_key(c, value);
+}
+
+/*
+ * set_value --
+ *     Wrapper providing the correct typing for the WT_CURSOR::set_value variadic argument.
+ */
+static void
+set_value(TEST_OPTS *opts, WT_CURSOR *c, uint64_t value)
+{
+    if (opts->table_type == TABLE_FIX)
+        c->set_value(c, (uint8_t)value);
+    else
+        c->set_value(c, value);
+}
+
+/*
+ * main --
+ *     Test's entry point.
+ */
+int
+main(int argc, char *argv[])
+{
+    TEST_OPTS *opts, _opts;
+    WT_CURSOR *cursor;
+    WT_SESSION *session;
+    clock_t ce, cs;
+    pthread_t id[NUM_THREADS];
+    int i, ret;
+    char tableconf[128];
+
+    opts = &_opts;
+    memset(opts, 0, sizeof(*opts));
+    opts->nthreads = NUM_THREADS;
+    opts->table_type = TABLE_ROW;
+    testutil_check(testutil_parse_opts(argc, argv, opts));
+    testutil_make_work_dir(opts->home);
+
+    testutil_check(
+      wiredtiger_open(opts->home, NULL, "create,cache_size=3G,statistics=(fast)", &opts->conn));
+    testutil_check(opts->conn->open_session(opts->conn, NULL, NULL, &session));
+    testutil_check(__wt_snprintf(tableconf, sizeof(tableconf),
+      "key_format=%s,value_format=%s,leaf_page_max=32k,memory_page_image_max=50MB",
+      opts->table_type == TABLE_ROW ? "Q" : "r", opts->table_type == TABLE_FIX ? "8t" : "Q"));
+    testutil_check(session->create(session, opts->uri, tableconf));
+
+    __wt_random_init_seed((WT_SESSION_IMPL *)session, &rnd);
+
+    cs = clock();
+
+    /* Multithreaded insert */
+    for (i = 0; i < (int)opts->nthreads; ++i)
+        testutil_check(pthread_create(&id[i], NULL, thread_insert_race, opts));
+
+    while (--i >= 0)
+        testutil_check(pthread_join(id[i], NULL));
+
+    /* Reopen connection for WT_SESSION::verify. It requires exclusive access to the file. */
+    testutil_check(opts->conn->close(opts->conn, NULL));
+    opts->conn = NULL;
+    testutil_check(wiredtiger_open(opts->home, NULL,
+      "create,cache_size=4G,eviction=(threads_max=1),statistics=(fast)", &opts->conn));
+
+    /* Validate */
+    testutil_check(opts->conn->open_session(opts->conn, NULL, NULL, &session));
+    testutil_check(session->verify(session, opts->uri, NULL));
+
+    testutil_check(session->open_cursor(session, opts->uri, NULL, NULL, &cursor));
+    i = 0;
+    while ((ret = cursor->next(cursor)) == 0)
+        i++;
```

Comment:
This is really a `for()` loop?

Reply:
Good point. Changed it 60e8dfc16161638548a0dd408ac335076dfb70ef